techmatters/terraso-mobile-client
# feat: offline-mode soil updates

## Description

Implement and unit test client-side versions of the four soil data mutations: `updateSoilData`, `deleteSoilDataDepthInterval`, `updateSoilDataDepthInterval`, and `updateDepthDependentSoilData`. When the offline feature flag is turned on, the client will use the local logic rather than making graphQL calls, and the resulting data will be ingested by the redux state.

### Checklist

### Related Issues
Fixes #2277

### Verification steps

Open the app with the offline feature flag turned on. Verify that all soil data actions (editing site settings, depth intervals, taking readings etc) behave as expected.
